internal/core/input.rs

```diff
@@ -176,6 +184,30 @@ pub struct KeyboardModifiers {
     pub shift: bool,
 }
 
+impl KeyboardModifiers {
+    /// Updates a flag of the modifiers if the key of the given text is pressed.
+    /// Returns an updated modifier if detected; None otherwise;
+    pub fn state_update(mut self, pressed: bool, text: &SharedString) -> Option<Self> {
+        if let Some(key_code) = text.chars().next() {
```

Comment:
That itself would not be enough. Right now the public API only allows for supplying a single char, so this code is fine. But in the future if we provide an entire string, then it could be like this:

`<control>foo<shift>bar`

and the `shift` modifier should only be set for `bar`, not for `foo`. So the modifiers need to not only be filtered but separate key events would need to be dispatched internally for the sub-strings.

I didn't want to go all the way with the complexity of that when right now the patch is already quite big and the interface doesn't allow for this scenario.

If you think that the code should already in this patch be generic, then I can change it.

Reply:
Indeed.